File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    if s[0]==')': return False
    answer = True
    
    arr = []
    
    for i in s:
        if i=='(':
            arr.append(i)
        else:  # i == ')'
            if len(arr) == 0: return False
            else: arr.pop()
    
    
    if len(arr) != 0: return False

    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        //처음부터 이상하게 열림
        if(s.charAt(0) == ')') return false;
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        for(char c : s.toCharArray()) {
            switch(c){
                case '(':
                    stack.push(c);
                    break;
                case ')':
                    if(!stack.isEmpty()){
                        stack.pop();//근데 이러면 ())도 통과되어버림
                    } else{
                        return false;
                    }
                    break;
            }
        }
        return stack.isEmpty();
    }
}